Cython/Compiler/Buffer.py

```diff
@@ -88,7 +88,7 @@ def decvar(type, prefix):
 
             auxvars = ((PyrexTypes.c_pyx_buffer_nd_type, Naming.pybuffernd_prefix),
                        (PyrexTypes.c_pyx_buffer_type, Naming.pybufferstruct_prefix))
-            pybuffernd, rcbuffer = [decvar(type, prefix) for (type, prefix) in auxvars]
+            pybuffernd, rcbuffer = (decvar(type, prefix) for (type, prefix) in auxvars)
```

Comment:
I'm not really convinced by changing most of these from list comprehensions - the advantage of a generator expression is really when you don't want the memory cost of creating the full list. But everything here is short so list comprehensions are probably just more efficient